draco/analysis/mapmaker.py

```diff
@@ -282,18 +317,114 @@ def _solve_m(self, m, f, v, Ni):
 
         # For large ntel it's quickest to solve in the standard Wiener filter way
         if bt.ntel > bt.nsky:
-            Ci = np.diag(1.0 / S_diag) + np.dot(bth, bmt)  # Construct the inverse covariance
-            a_dirty = np.dot(bth, vt)  # Find the dirty map
+            Ci = np.diag(1.0 / S_diag) + np.dot(beam_conj, beam)  # Construct the inverse covariance
+            a_dirty = np.dot(beam_conj, vec)  # Find the dirty map
             a_wiener = la.solve(Ci, a_dirty, sym_pos=True)  # Solve to find C vt
 
         # If not it's better to rearrange using the results for blockwise matrix inversion
         else:
-            pCi = np.identity(bt.ntel) + np.dot(bmt * S_diag[np.newaxis, :], bth)
-            v_int = la.solve(pCi, vt, sym_pos=True)
-            a_wiener = S_diag * np.dot(bth, v_int)
+            pCi = np.identity(vec.shape[0]) + np.dot(beam * S_diag[np.newaxis, :], beam_conj)
+            v_int = la.solve(pCi, vec, sym_pos=True)
+            a_wiener = S_diag * np.dot(beam_conj, v_int)
+
+        return a_wiener
+
+
+class DirtyMapMakerSVD(BaseMapMaker):
+    """Generate a dirty map in SVD basis.
+
+    Notes
+    -----
+
+    The dirty map is produced by generating a set of :math:`a_{lm}` coefficients
+    using
+
+    .. math:: \hat{\mathbf{a}} = \bar{\mathbf{B}^{+}}  \bar{\mathbf{v}}
+
+    and then performing the spherical harmonic transform to get the sky intensity.
+    """
+
+    def _solve_m(self, m, f, vec, Ni):
+
+        bt = self.beamtransfer
+
+        # Get the svd beamtransfer matrix for a specific m and f
+        beam = bt.beam_svd(m, f)
+        # Create alm array
+        a = np.zeros((bt.telescope.num_pol_sky, bt.telescope.lmax + 1), dtype=vec.dtype)
+        # Get the significant svd modes and the frequency bounds
+        svnum, svbounds = bt._svd_num(m)
+
+        if svnum[f] < 1:
+            return a
+
+        # Get the svd vector for this frequency
+        fvec = vec[svbounds[f]:svbounds[f+1]]
+        # Get the inverse beam with the significant svd modes but trim off any l < m.
+        beam_svnum = beam[:svnum[f], :, m:]
+        beam_svnum = beam_svnum.reshape(svnum[f], -1)
+
+        # Construct beam-conjugated matrices
+        beam_svnum_conj = beam_svnum.T.conj()
+
+        # Solve for the dirty map alms
+        a = np.dot(beam_svnum_conj, fvec)
+        a = a.reshape(bt.telescope.num_pol_sky, -1)
+
+        return a
+
+
+class WienerMapMakerSVD(WienerMapMaker):
```

Comment:
Does this really need to be a separate sub class? It seems to me that you could probably merge them entirely. You just need to massage the projection matrices to be the same form. Same for `DirtyMapMaker`.

Reply:
Hmmmm, well I thought about this, we could probably merge them, but we would have to have some if else structures. Some of them include:

- 'beam_m' versus  'beam_svd'

- line 356: `svnum, svbounds = bt._svd_num(m)' for svdmodes

- when dealing with svdmodes we need to slice the beam transfer matrices and the data vector according to `svnum[f]`

- For Svd modes we need to create a noise covriance matrix `Ni` that is unity to merge the code.

Any comments?